### DISCUSSION TITLE
Support for sqlite backend

### DISCUSSION THREAD

**Discussion by luaneko** [umami-software/umami#930] [Ideas]
Hi! Thanks for making this great analytics app.

It would be great to see an SQLite database backend in addition to MySQL and PostgreSQL. SQLite is suitable for light workloads or when installing a database server isn't necessarily feasible.

From what I've observed, umami uses Prisma 2 for database connection which natively supports SQLite already. It would be a matter of simply adding a Prisma schema for it, modifying a few raw queries and distributing another Docker image.

**Comment by mikecao** [comment]
Yes, it would be possible, just not a high priority right now. I would welcome a PR from anyone who wants to tackle it.

**Reply by Maxime-J** [reply]
I have a working implementation, with appropriate date/time handling, following the same logic and structure as currently.
A matter of days before a PR :)

**Reply by kaichaosun** [reply]
Check out https://github.com/kaichaosun/swa

**Comment by leirock** [comment]
I notice that @philippdormann's [modded version of umami](https://github.com/philippdormann/umami-sqlite) supports a local SQLite database. But seems no PR was created?

**Comment by philippdormann** [comment]
hi @leirock 
yes, it does support SQLite and worked last time I checked.

I stopped working on this due to time constraints and the (for my use case too high to justify) memory usage of umami.
afaik there were some major changes in v1.17.0 in terms of Prisma/ Next.js version bumps. Those dependency bumps might require some changes for sqlite to work 🤷‍♂️

see https://github.com/mikecao/umami/pull/680#issuecomment-909228819 for the PR 😉

note: I might get around to a proper PR for sqlite support with the current version. recently got some more time :)

**Comment by leirock** [comment]
hi @philippdormann 
Thanks very much for your reply. It's good news to hear that you got some more time recently. Cheers! 😂

**Comment by Maxime-J** [comment]
Hello!
I'd like to work on that if wanted.
Find it weird to have a working sqlite support without touching lib/queries.js

**Comment by calumk** [comment]
I know this is revisited every now and again, but +1 for sqlite support.
Would be wonderful for self-hosting more easily.

**Comment by alphaleadership** [comment]
any update

**Comment by voidborn-one** [comment]
SQLite would be great for small scale self-hosted websites to minimize the footprints

**Comment by przemoc** [comment]
Just letting others know about a comment from Maxime (maintainer of https://github.com/Maxime-J/umami-sqlite) in PR:
- https://github.com/umami-software/umami/pull/1292#issuecomment-2663116536

triggered by my info left there regarding relatively recent Prisma update potentially making some things easier (or so I thought).

**Comment by LeonardoIz** [comment]
then sqlite will be added?

**Comment by kaichaosun** [comment]
A lightweight, single-binary web analytics tool built with Rust and SQLite. No external dependencies, no cloud services — just deploy and go. https://github.com/kaichaosun/swa
